### DISCUSSION TITLE
Could a tiktok user's secUid ever change in its lifetime?

### DISCUSSION THREAD

**Discussion by ZYinMD** [davidteather/TikTok-Api#658] [Q&A]
I know a tiktok user's `id` (integer) won't ever change, while `@uniqueId` can definitely change, but what about `secUid`? If secUid also never changes, what's the purpose of it?

**Comment by liamengland1** [comment]
No it doesn't

**Reply by ZYinMD** [reply]
If it can never change, then why did TikTok create it in the first place? Both the id and the secUid can be used to identify a user, and the secUid is not more secret than the id.

**Reply by liamengland1** [reply]
How am I supposed to know that? Do I work at tiktok? :-)

**Reply by ZYinMD** [reply]
Well, it's very possible that someone more experienced than me may know, it could be a standard practice in API designing. For instance, maybe it prevents certain attacks which increments the id by 1 to get to the next user. I don't know, but I've read about things like it in other places.

**Comment by adiantek** [comment]
They changed ID and secUID :o
6653803073764855814 - old ID
6653806073764855814 - new ID
`xS4wLjABAAAAYCsPXEsmmsB9v2JCv6vSEW-CXqSkfNL88ME6qw8Md-d8F_zkcrHQu0b3tz5mlQJa` - old secUID
`MS4wLjABAAAAYCsPXEsmmsB9v2JCv6vSEW-CXqSkfNL88ME6qw8Md-d8F_zkcrHQu0b3tz5mlQJa` - new secUID
It happened only to my friend's account, don't know why. I tested other accounts and without changes.
When I make item_list (old ID and old secUID) request I'm getting an empty item list XD

**Reply by ZYinMD** [reply]
That is very hard to believe, a single digit change. I don't know your setup, but I'd rather believe it's somehow a human error made by you.

**Reply by adiantek** [reply]
He wrote with support about verify account and change phone number. I had a statistics crawler and before I wrote this I saw that there are 0 videos. You can check it and make item_list request - you will get 0 videos, but if you make a human error you will get an invalid entity exception (secUID not found). But still - don't know why they changed one digit of ID and secUID.
